### PR TITLE
Specify paths and shell on Windows

### PR DESCRIPTION
**Pull request by davidwengier**
My default terminal in VS Code is bash, but I'm on Windows so VS Code uses windows slashes, and generally results in pain. These changes fix things by forcing `cmd.exe` usage on Windows, and specifying the absolute path of some commands.

Would love someone to validate that I haven't broken anything, especially on non-Windows (@333fred)

### CODE REVIEW

File: .vscode/tasks.json [hunk 1]
```diff
@@ -102,7 +112,7 @@
         "command": "./build.sh",
         "type": "shell",
         "windows": {
-          "command": "./build.cmd"
+          "command": "${workspaceFolder}/build.cmd"
```
Comment (333fred) [inline .vscode/tasks.json:115]: Why the differences between workspaceFolder and workspaceRoot?

**Review by 333fred — COMMENTED** [review]
I'll try to check this out and test later today.

File: .vscode/tasks.json (hunk 1 above)
Comment (333fred) [reply .vscode/tasks.json:115]: Actually, I think these should all be workspaceFolder. https://code.visualstudio.com/docs/editor/variables-reference#_why-isnt-workspaceroot-documented

File: .vscode/tasks.json [hunk 2]
```diff
@@ -2,6 +2,16 @@
   // See https://go.microsoft.com/fwlink/?LinkId=733558
   // for the documentation about the tasks.json format
   "version": "2.0.0",
+  "windows": {
```
Comment (333fred) [inline .vscode/tasks.json:5]: What does this actually control? Does it control what terminal is used when I hit `` ctrl+` `` on Windows? Or just what shell these tasks run in?

File: .vscode/tasks.json (hunk 1 above)
Comment (davidwengier) [reply .vscode/tasks.json:115]: Whoops. The inconsistency was just that this was done in two parts, over two days, on two machines, and I picked a different thing to copy and paste each time it seems.

Great catch on the `workspaceRoot` being deprecated though, I'll update all of the existing uses too.

File: .vscode/tasks.json (hunk 2 above)
Comment (davidwengier) [reply .vscode/tasks.json:5]: Just what shell is used to run tasks in.

**Review by 333fred — APPROVED** [review]

**Pull request merged**